### PR TITLE
Slightly simplify OpenID modules

### PR DESCRIPTION
**Pull request by okurz**
* Simplify WebAPI::Auth::OpenID
* Simplify WebAPI::Controller::Session
* Remove unused import in OpenQA::WebAPI::Controller::Session

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/Session.pm [hunk 1]
```diff
@@ -16,57 +16,29 @@
...
-            $self->render(text => 'Bad CSRF token!', status => 403);
-        }
-        else {
-            $self->render(text => "Forbidden", status => 403);
-        }
-    }
-    else {
-        $self->redirect_to('login');
-    }
+    $self->redirect_to('login') and return unless $self->current_user;
```
Comment (kraih) [inline lib/OpenQA/WebAPI/Controller/Session.pm:29]: I thought we agreed that `return $self->redirect_to('login') unless $self->current_user` looks better? Old habit?

File: lib/OpenQA/WebAPI/Controller/Session.pm [hunk 2]
```diff
@@ -16,57 +16,29 @@
...
```
Comment (kraih) [inline lib/OpenQA/WebAPI/Controller/Session.pm:38]: This should probably be a `return undef`. (Same for the cases below)

**Comment by okurz** [comment]
Yes, I tested manually as well, e.g. use `httpsonly = 0` in etc/openqa/openqa.ina and ensure I can login as a user as well as stay logged in authenticated on browsing to different pages. I saw that the automatic coverage is not complete. This might be something to improve as followup.

File: lib/OpenQA/WebAPI/Controller/Session.pm (hunk 2 above)
Comment (okurz) [reply lib/OpenQA/WebAPI/Controller/Session.pm:38]: I agree but I wanted to keep the code as close as possible for a start. I will add a commit on top to change that.

File: lib/OpenQA/WebAPI/Controller/Session.pm [hunk 3]
```diff
@@ -76,10 +48,7 @@ sub destroy {
     my $auth_method = $self->app->config->{auth}->{method};
     my $auth_module = "OpenQA::WebAPI::Auth::$auth_method";
     eval { $auth_module->import('auth_logout'); };
-    if (!$@) {
-        auth_logout($self);
-    }
-
+    auth_logout($self) unless $@;
```
Comment (kraih) [inline lib/OpenQA/WebAPI/Controller/Session.pm:51]: Importing in the middle of code that runs over and over is rather ugly. How about something like this?
```perl
if (my $sub = $auth_module->can('auth_logout')) { $self->$sub) }
```

**Comment by okurz** [comment]
I will include the import changes and also test

File: lib/OpenQA/WebAPI/Controller/Session.pm (hunk 3 above)
Comment (okurz) [reply lib/OpenQA/WebAPI/Controller/Session.pm]: done

File: lib/OpenQA/WebAPI/Controller/Session.pm (hunk 2 above)
Comment (okurz) [reply lib/OpenQA/WebAPI/Controller/Session.pm]: done

**Pull request merged**